pkg/operator/ceph/object/spec.go

```
@@ -492,29 +493,48 @@ func getLabels(name, namespace string, includeNewLabels bool) map[string]string
 	// Also, we don't want to change the SCC for fsGroup to RunAsAny since it has a major broader impact
 	// Let's open the permissions a bit more so that everyone can read the cert.
 	userReadOnly := int32(0444)
 	var secretVolSrc *v1.SecretVolumeSource
-	if objSpec.Gateway.SSLCertificateRef != "" {
+	if c.store.Spec.Gateway.SSLCertificateRef != "" {
 		secretVolSrc = &v1.SecretVolumeSource{
-			SecretName: objSpec.Gateway.SSLCertificateRef,
-			Items: []v1.KeyToPath{
+			SecretName: c.store.Spec.Gateway.SSLCertificateRef,
+		}
+		if c.rgwTLSSecretType() == "" {
+			return nil, errors.New("no TLS certificates found")
+		} else if c.rgwTLSSecretType() == v1.SecretTypeOpaque {
+			secretVolSrc.Items = []v1.KeyToPath{
 				{Key: certKeyName, Path: certFilename, Mode: &userReadOnly},
-			}}
-	} else if objSpec.GetServiceServingCert() != "" {
+			}
+		} else if c.rgwTLSSecretType() == v1.SecretTypeTLS {
+			secretVolSrc.Items = []v1.KeyToPath{
+				{Key: v1.TLSCertKey, Path: certFilename, Mode: &userReadOnly},
+				{Key: v1.TLSPrivateKeyKey, Path: certKeyFileName, Mode: &userReadOnly},
+			}
+		}
+	} else if c.store.Spec.GetServiceServingCert() != "" {
 		secretVolSrc = &v1.SecretVolumeSource{
-			SecretName: objSpec.GetServiceServingCert(),
+			SecretName: c.store.Spec.GetServiceServingCert(),
 			Items: []v1.KeyToPath{
-				{Key: "tls.crt", Path: certFilename, Mode: &userReadOnly},
-				{Key: "tls.key", Path: certKeyFileName, Mode: &userReadOnly},
+				{Key: v1.TLSCertKey, Path: certFilename, Mode: &userReadOnly},
+				{Key: v1.TLSPrivateKeyKey, Path: certKeyFileName, Mode: &userReadOnly},
 			}}
 	} else {
 		return nil, errors.New("no TLS certificates found")
 	}
 
 	return secretVolSrc, nil
 }
+
+func (c *clusterConfig) rgwTLSSecretType() v1.SecretType {
+	ctx := context.TODO()
+	rgwTlsSecret, _ := c.context.Clientset.CoreV1().Secrets(c.clusterInfo.Namespace).Get(ctx, c.store.Spec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+	if rgwTlsSecret != nil {
```

Why not return an error since the caller can handle errors?